community/detectors/apache_http_server_cve_2021_41773/src/main/java/com/google/tsunami/plugins/detectors/rce/cve_2021_25646/ApacheHttpServerVulnDetector.java

```diff
@@ -0,0 +1,115 @@
+    return DetectionReportList.newBuilder()
+        .addAllDetectionReports(
+            matchedServices.stream()
+                .filter(NetworkServiceUtils::isWebService)
+                .filter(this::isServiceVulnerable)
+                .map(networkService -> buildDetectionReport(targetInfo, networkService))
+                .collect(toImmutableList()))
+        .build();
+  }
+
+  private boolean isServiceVulnerable(NetworkService networkService) {
+    String targetUri =
+        NetworkServiceUtils.buildWebApplicationRootUrl(networkService) + "cgi-bin/.%2e/";
+    try {
+      HttpResponse response = httpClient.send(get(targetUri).withEmptyHeaders().build(),
+          networkService);
+      try {
+        Optional<String> server = response.headers().get("Server");
+        Optional<String> body = response.bodyString();
+        if (server.isPresent() && server.get().contains("Apache/2.4.49") && body.isPresent()
+            && !body.get().contains("You don't have permission to access this resource.")) {
+          return true;
+        }
+      } catch (Throwable t) {
+        logger.atInfo().log("Failed to parse cores response json");
+      }
+    } catch (IOException e) {
+      logger.atWarning().withCause(e).log("Unable to query '%s'.", targetUri);
+    }
+    return false;
+  }
+
+  public DetectionReport buildDetectionReport(
+      TargetInfo targetInfo, NetworkService vulnerableNetworkService) {
+    return DetectionReport.newBuilder()
+        .setTargetInfo(targetInfo)
+        .setNetworkService(vulnerableNetworkService)
+        .setDetectionTimestamp(Timestamps.fromMillis(Instant.now(utcClock).toEpochMilli()))
+        .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+        .setVulnerability(
+            Vulnerability.newBuilder()
+                .setMainId(
+                    VulnerabilityId.newBuilder().setPublisher("TSUNAMI_COMMUNITY")
+                        .setValue("CVE-2021-41773"))
+                .setSeverity(Severity.HIGH)
+                .setTitle("Apache HTTP Server 2.4.49 Path traversal and disclosure vulnerability")
+                .setDescription(
+                    "A flaw was found in a change made to path normalization in Apache HTTP Server 2.4.49. An attacker could use a path traversal attack to map URLs to files outside the expected document root."
+                        + "If files outside of the document root are not protected by \"require all denied\" these requests can succeed. Additionally this flaw could leak the source of interpreted files like CGI scripts."
+                        + "This issue is known to be exploited in the wild."
+                        + "This issue only affects Apache 2.4.49 and not earlier versions."
+                        + "https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-41773 "
+                        + "https://httpd.apache.org/security/vulnerabilities_24.html")
```

Comment:
Can you please set the `recommendation` field as well?

Reply:
added,done.